internal/provider/schema_registry_resource.go

```diff
@@ -0,0 +1,245 @@
+package provider
```

Comment:
these are pretty much boilerplate stuff that are almost identical to `virtual_cluster_resource`. However, they will diverge in the future as we add schema registry specific configurations